API/hermes_abi/hermes_abi.h

```diff
@@ -0,0 +1,400 @@
+  HermesABIString (*clone_string)(HermesABIContext *, HermesABIString);
+  HermesABISymbol (*clone_symbol)(HermesABIContext *, HermesABISymbol);
+  HermesABIObject (*clone_object)(HermesABIContext *, HermesABIObject);
+  HermesABIBigInt (*clone_big_int)(HermesABIContext *, HermesABIBigInt);
+
+  /// Check if the given buffer contains Hermes bytecode.
+  bool (*is_hermes_bytecode)(const uint8_t *, size_t);
+
+  /// Evaluate the given JavaScript buffer with an associated source URL in the
+  /// given context, and return the result.
+  HermesABIValueOrError (*evaluate_javascript)(
+      HermesABIContext *,
+      HermesABIBuffer *,
+      const char *,
+      size_t);
+
+  HermesABIObject (*get_global_object)(HermesABIContext *);
+  HermesABIStringOrError (
+      *create_string_from_ascii)(HermesABIContext *, const char *, size_t);
+  HermesABIStringOrError (
+      *create_string_from_utf8)(HermesABIContext *, const uint8_t *, size_t);
+
+  HermesABIObjectOrError (*create_object)(HermesABIContext *);
+
+  HermesABIBoolOrError (*has_object_property_from_string)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIString);
+  HermesABIBoolOrError (*has_object_property_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIPropNameID);
+
+  HermesABIValueOrError (*get_object_property_from_string)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIString);
+  HermesABIValueOrError (*get_object_property_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIPropNameID);
+  HermesABIVoidOrError (*set_object_property_from_string)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIString,
+      const HermesABIValue *);
+  HermesABIVoidOrError (*set_object_property_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIPropNameID,
+      const HermesABIValue *);
+
+  HermesABIArrayOrError (
+      *get_object_property_names)(HermesABIContext *, HermesABIObject);
+
+  HermesABIArrayOrError (*create_array)(HermesABIContext *, size_t);
+  size_t (*get_array_length)(HermesABIContext *, HermesABIArray);
+  HermesABIValueOrError (
+      *get_array_value_at_index)(HermesABIContext *, HermesABIArray, size_t);
+  HermesABIVoidOrError (*set_array_value_at_index)(
+      HermesABIContext *,
+      HermesABIArray,
+      size_t,
+      const HermesABIValue *);
+
+  HermesABIPropNameIDOrError (*create_prop_name_id_from_ascii)(
+      HermesABIContext *,
+      const char *,
+      size_t);
+  HermesABIPropNameIDOrError (*create_prop_name_id_from_utf8)(
+      HermesABIContext *,
+      const uint8_t *,
+      size_t);
+  HermesABIPropNameIDOrError (
+      *create_prop_name_id_from_string)(HermesABIContext *, HermesABIString);
+  HermesABIPropNameIDOrError (
+      *create_prop_name_id_from_symbol)(HermesABIContext *, HermesABISymbol);
+  bool (*compare_prop_name_id)(
```

Comment:
Usually term "compare" is used in C++ when we expect to get -1, 0, or 1.
They use term "equals" to return bool. E.g. [std::equal](https://en.cppreference.com/w/cpp/algorithm/equal)

Reply:
Below you use term `strict_equals`.

Reply:
On the other hand, this is the current naming convention used in JSI.

Reply:
You are changing some names anyway. Using `strict_equals_prop_name_id` will align it with other functions and clean up the semantics by using the verb `equals`.

Reply:
I am with @vmoroz on this one